django_file_form/views/s3multipart.py

```diff
@@ -0,0 +1,184 @@
+                    aws_access_key_id=cls.aws_access_key_id or
+                    lookup_env(['AWS_S3_ACCESS_KEY_ID', 'AWS_ACCESS_KEY_ID']),
+                    aws_secret_access_key=cls.aws_secret_access_key or
+                    lookup_env(
+                        ['AWS_S3_SECRET_ACCESS_KEY', 'AWS_SECRET_ACCESS_KEY']))
+            except:
+                time.sleep(0.01)
+
+    @classmethod
+    def exists(cls, client, bucket_name, name):
+        """
+        Check if key already exists bucket,
+
+        Code adapted from storage.backends.s3boto3
+        """
+        try:
+            client.head_object(Bucket=bucket_name, Key=name)
+            return True
+        except ClientError:
+            return False
+
+    @classmethod
+    def get_alternative_name(cls, file_root, file_ext):
+        """
+        Return an alternative filename, by adding an underscore and a random 7
+        character alphanumeric string (before the file extension, if one
+        exists) to the filename.
+
+        Code adapted from django.storage.get_alternative_name
+        """
+        return f'{file_root}_{get_random_string(7)}{file_ext}'
+
+    @classmethod
+    def get_available_name(cls, client, bucket_name, name, max_length=None):
+        """
+        Return a filename that's free on the target storage system and
+        available for new content to be written to.
+
+        Code adapted from django.storage.get_available_name
+        """
+        dir_name, file_name = os.path.split(name)
+        file_root, file_ext = os.path.splitext(file_name)
+        # If the filename already exists, generate an alternative filename
+        # until it doesn't exist.
+        # Truncate original name if required, so the new filename does not
+        # exceed the max_length.
+        while cls.exists(client, bucket_name, name) or (max_length and
+                                                        len(name) > max_length):
+            # file_ext includes the dot.
+            name = os.path.join(dir_name,
+                                cls.get_alternative_name(file_root, file_ext))
+            if max_length is None:
+                continue
+            # Truncate file_root if max_length exceeded.
+            truncation = len(name) - max_length
+            if truncation > 0:
+                file_root = file_root[:-truncation]
+                # Entire file_root was truncated in attempt to find an available filename.
+                if not file_root:
+                    raise SuspiciousFileOperation(
+                        'Storage can not find an available filename for "%s". '
+                        'Please make sure that the corresponding file field '
+                        'allows sufficient "max_length".' % name)
+                name = os.path.join(
+                    dir_name, cls.get_alternative_name(file_root, file_ext))
+        return name
+
+    @classmethod
+    def createMultipartUpload(cls, request):
```

Comment:
Could you use snake case for this (`create_multipart_upload`)? Also applies to `getPartsOrAbortUpload` and other methods.

Reply:
Will do later but you could also push to our branch if it is easier for you.

Reply:
Done. Hopefully did not introduce any error.